### PR TITLE
Access nodes to be used

### PR DESCRIPTION
**Pull request by kape1395**

### CODE REVIEW

File: packages/chain/node.go [hunk 1]
```diff
@@ -222,11 +236,14 @@ func New(
...
 		publishingTXes:         map[iotago.TransactionID]context.CancelFunc{},
 		procCache:              processors.MustNew(processorConfig),
+		configUpdatedCh:        make(chan *configUpdate, 1),
 		listener:               listener,
 		accessLock:             &sync.RWMutex{},
 		activeCommitteeDKShare: nil,
 		activeCommitteeNodes:   []*cryptolib.PublicKey{},
-		activeAccessNodes:      []*cryptolib.PublicKey{},
+		activeAccessNodes:      nil,                // updated bellow.
+		accessNodesFromNode:    accessNodesForNode, // TODO: Update it on event.
```
Comment (jorgemmsilva) [inline packages/chain/node.go]:
```suggestion
		accessNodesFromNode:    accessNodesForNode,
```
we can probably remove the TODO

**Review by jorgemmsilva — APPROVED** [review]

File: packages/chains/chains.go [hunk 2]
```diff
@@ -110,6 +111,14 @@ func (c *Chains) Run(ctx context.Context) error {
 	}
 	c.ctx = ctx
 
+	c.chainRecordRegistryProvider.Events().ChainRecordModified.Attach(event.NewClosure(func(event *registry.ChainRecordModifiedEvent) {
+		c.mutex.Lock()
```
Comment (muXxer) [inline packages/chains/chains.go]: Wouldn't a RLock be sufficient?
And you can move the defer directly after the RLock().

**Review by muXxer — CHANGES_REQUESTED** [review]

File: packages/chain/node.go (hunk 1 above)
Comment (kape1395) [reply packages/chain/node.go]: Thanks!

File: packages/chains/chains.go (hunk 2 above)
Comment (kape1395) [reply packages/chains/chains.go]: Good point, thanks.

**Review by muXxer — APPROVED** [review]

**Pull request merged**